contracts/core/OriginCore.sol

```diff
@@ -353,9 +372,20 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
             _s,
             verifiedWeight,
             MetaBlock.requiredWeightForSuperMajority(
-                    stake.totalWeightAtHeight(height)
+                stake.totalWeightAtHeight(height)
             )
         );
+
+        /* Meta-block should only be committed once. */
+        if (_kernelHash == openKernelHash) {
+            commitMetaBlock(
+                _kernelHash,
+                _transitionHash,
+                height,
+                verifiedWeight
+            );
+        }
+
```

Comment:
`stake.totalWeightAtHeight(height)` is expensive. 
On L375, we already have the calculated value. We can use that and check the condition `if (_verifiedWeight >= requiredWeight) ` before calling `commitMetaBlock `